### ISSUE TITLE
[Rule] Java: SQLi in a query that uses CriteriaBuilder

### ISSUE THREAD

**Issue by N3tn1la** [returntocorp/sgrep-rules#2831]
**Describe the bug**
The rule java.lang.security.audit.formatted-sql-string.formatted-sql-string give a false positive results when analysing sources with a CriteriaBuilder.
I think that append to the rule "pattern-non" could fix that problem

**To Reproduce**
Sample code to reproduce this behavior.
``` Java
@SpringBootApplication
public class SpringMain {
	public static void main(String[] args) {
		SpringApplication.run(SpringMain.class, args);
	}
}

@RestController
@RequestMapping("/student")
public class StudentController {

	@Autowired
	private StudentService studentService;

	@PostMapping("/create")
	public Student createStudent(@RequestBody Student student) {
		Student createResponse = studentService.save(student);
		return createResponse;
	}
	
	@GetMapping("/allstudents")
	public List<Student> getStudents() {
		List<Student> getReponse = studentService.getStudents();
		return getReponse;
	}
}

@Entity
public class Student {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column(name = "name")
	private String name;

	@Column(name = "roll_number")
	private String rollNumber;

	@Column(name = "university")
	private String university;
	
//getter & setter

}

@Service
public class StudentServiceImpl implements StudentService {

	@PersistenceContext
	private EntityManager entityManager;

	@Transactional
	public Student save(Student student) {
		entityManager.persist(student);
		return student;
	}

	@Transactional
	public List<Student> getStudents() {

		CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
		CriteriaQuery<Student> criteriaQuery = criteriaBuilder.createQuery(Student.class);
		Root<Student> studentRoot = criteriaQuery.from(Student.class);
		criteriaQuery.orderBy(criteriaBuilder.desc(studentRoot.get("name")));
		List<Student> students = entityManager.createQuery(addWhere("Alex", criteriaQuery)).getResultList();
		return students;
	}

    public List<Student> addWhere(String name, CriteriaQuery Query)
    {
        return Query.where(criteriaBuilder.equal(studentRoot.get("name"), name ));
    }

}
```

**Expected behavior**
I expected to happen that all will be ok, without error. I think problem is in createQuery, that can have String inside and it's not bad if Criteria Query continue exist.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**

**Comment by inkz** [comment]
Hi @danchik1155 thanks for reporting this! Going to work on it soon :pray:

**Comment by inkz** [comment]
@danchik1155 PR with updated rule created: https://github.com/returntocorp/semgrep-rules/pull/2851
